db/python/layers/project_insights.py

```diff
@@ -301,367 +332,601 @@ async def _stats_es_indexes_query(
+                project_ids, sequencing_types
+            ),
+            self._total_sequencing_groups_by_project_id_and_seq_fields(
+                project_ids, sequencing_types
+            ),
+            self._crams_by_project_id_and_seq_fields(project_ids, sequencing_types),
+            self._latest_annotate_dataset_by_project_id_and_seq_type(
+                project_ids, sequencing_types
             ),
-            self._stats_crams_query(projects, sequencing_types),
-            self._stats_annotate_dataset_query(
-                projects, sequencing_types
+            self._latest_es_indices_by_project_id_and_seq_type_and_stage(
+                project_ids,
+                sequencing_types,
             ),
-            self._stats_es_indexes_query(projects, sequencing_types,),
+        )
+
+        # Get the sequencing groups for each of the analyses in the grouped analyses rows
+        analysis_sequencing_groups = await self.get_analysis_sequencing_groups(
+            [
+                latest_annotate_dataset_by_project_id_and_seq_type,
+                latest_es_indices_by_project_id_and_seq_type_and_stage,
+            ]
+        )
+
+        sequencing_technologies = await SequencingTechnologyTable(
+            self._connection
+        ).get()
+
+        # Get all possible permutations of the projects, sequencing types, and sequencing technologies
+        permutations = itertools.product(
+            projects, sequencing_types, sequencing_technologies
         )
 
         response = []
-        for pid in projects:
-            project = await ptable.get_and_check_access_to_project_for_id(self.author, pid, readonly=True)
-            for sequencing_type in sequencing_types:
-                crams_in_project_and_sequencing_type = [
-                    cram['sequencing_group_id']
-                    for cram in crams
-                    if cram['project'] == project.id
-                    and cram['sequencing_type'] == sequencing_type
+        for project, sequencing_type, sequencing_technology in permutations:
+            total_sequencing_groups = (
+                total_sequencing_groups_by_project_id_and_seq_fields.get(
+                    (project.id, sequencing_type, sequencing_technology), 0
+                )
+            )
+            if total_sequencing_groups == 0:
+                continue
+
+            crams_in_project_with_sequencing_fields = (
+                crams_by_project_id_and_seq_fields[
+                    (project.id, sequencing_type, sequencing_technology)
                 ]
-                latest_annotate_dataset_id = (
-                    self.get_val_for_project_and_sequencing_type(
-                        project.id, sequencing_type, 'id', latest_annotate_dataset
-                    )
+            )
+
+            if sequencing_technology == "short-read":
+                latest_annotate_dataset = AnalysisStatsInternal(
```

Comment:
This function is long, and nested makes it harder to read, consider moving this construction to it's own function